Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -451,22 +452,23 @@ PPCDebugInterface::GetMemoryAddressFromInstruction(const std::string& instructio
   if (!std::regex_search(instruction, match, re))
     return std::nullopt;
 
-  // Output: match.str(1): negative sign for offset or no match. match.str(2): 0xNNNN, 0, or
-  // rNN. Check next for 'r' to see if a gpr needs to be loaded. match.str(3): will either be p,
-  // toc, or NN. Always a gpr.
-  const std::string offset_match = match.str(2);
-  const std::string register_match = match.str(3);
+  // match[1]: negative sign for offset or no match.
+  // match[2]: 0xNNNN, 0, or rNN. Check next for 'r' to see if a gpr needs to be loaded.
+  // match[3]: will either be p, toc, or NN. Always a gpr.
+  const std::string_view offset_match{match[2].first.operator->(), size_t(match[2].length())};
+  const std::string_view register_match{match[3].first.operator->(), size_t(match[3].length())};
```

Comment:
I haven't looked too much into your other PRs, but `match[2].first.operator->()` wasn't super obvious to me. The key thing is that [`std::smatch`](https://en.cppreference.com/w/cpp/regex/match_results) uses `std::string::const_iterator`, and [`std::ssub_match`](https://en.cppreference.com/w/cpp/regex/sub_match) inherits `std::pair` of two string iterators (with the `first` being the start and the `second` being the one past the end). So you need to convert the iterator into a pointer (because of a lack of support for [`std::string_view`'s iterator constructor](https://en.cppreference.com/w/cpp/string/basic_string_view/basic_string_view) as you mentioned).

And I guess `operator->()` is the best way of doing that? I feel like `&*match[2].first` would be a bit more obvious but maybe that has a downside I'm not aware of?

Reply:
https://github.com/dolphin-emu/dolphin/pull/11934 was supposed to introduce a function in the Common namespace that abstracted this esoteric behavior, but it never took off.  For me, it's just the principle; dereferencing then referencing an iterator to get a pointer out of it seems like code smell to me.  For the sake of readability, though, I suppose I can resort to the more recognizable pattern.  It probably optimizes to the exact same thing anyhow.

Reply:
This is all to optimize a horribly inefficient disassembly re-assembler that would be better off replaced anyhow.